lib/private/Files/Cache/QuerySearchHelper.php

```diff
@@ -196,23 +195,31 @@ public function searchInCaches(ISearchQuery $searchQuery, array $caches): array
 	}
 
 	/**
-	 * @return array{array<string, ICache>, array<string, IMountPoint>}
+	 * @return array{0?: array<string, ICache>, 1?: array<string, IMountPoint>}
 	 */
-	public function getCachesAndMountPointsForSearch(Root $root, string $path, bool $limitToHome = false): array {
+	public function getCachesAndMountPointsForSearch(IRootFolder $root, string $path, bool $limitToHome = false): array {
 		$rootLength = strlen($path);
-		$mount = $root->getMount($path);
-		$storage = $mount->getStorage();
+		if (method_exists($root, 'getMount')) {
```

Comment:
```suggestion
		$storage = null;
		if (method_exists($root, 'getMount')) {
```
Would feel more readable this way.